src/lib.rs

```diff
@@ -131,12 +146,13 @@ impl Connection {
             header_bytes[0] &= !PBF_MASK;
             header_bytes[0] |= (pbf as u8) << 4;
 
-            if MessageType::try_from(message_type) == Ok(MessageType::Data) {
-                let [low, high] = (chunk_length as u16).to_le_bytes();
-                header_bytes[2] = low;
-                header_bytes[3] = high;
-            } else {
-                header_bytes[3] = chunk_length as u8;
+            match message_type {
+                MessageType::Data => {
+                    let [low, high] = (chunk_length as u16).to_le_bytes();
+                    header_bytes[2] = low;
+                    header_bytes[3] = high;
```

Comment:
nit (just to save a line):
```suggestion
                    let chunk_le_bytes = (chunk_length as u16).to_le_bytes();
                    header_bytes[2..4].copy_from_slice(&chunk_le_bytes);
```